root/user/UserCollections.js

```diff
@@ -0,0 +1,265 @@
+/*
+ * @flow
+ * Copyright (C) 2019 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import some from 'lodash/some';
+import * as React from 'react';
+import {useTable} from 'react-table';
+
+import UserAccountLayout from '../components/UserAccountLayout';
+import {withCatalystContext} from '../context';
+import EntityLink from '../static/scripts/common/components/EntityLink';
+import {unwrapNl} from '../static/scripts/common/i18n';
+import {formatPluralEntityTypeName} from '../static/scripts/common/utility/formatEntityTypeName';
+import yesNo from '../static/scripts/common/utility/yesNo';
+import loopParity from '../utility/loopParity';
+
+type Props = {|
+  +$c: {user: EditorT} & CatalystContextT,
+  +collaborativeCollections: CollectionListT,
+  +ownCollections: CollectionListT,
+  +user: EditorT,
+|};
+
+type CollectionListT = {|
+  +[string]: $ReadOnlyArray<CollectionT>,
+|};
+
+const collectionsListTitle = {
+  area: N_l('Area Collections'),
+  artist: N_l('Artist Collections'),
+  event: N_l('Event Collections'),
+  instrument: N_l('Instrument Collections'),
+  label: N_l('Label Collections'),
+  place: N_l('Place Collections'),
+  recording: N_l('Recording Collections'),
+  release: N_l('Release Collections'),
+  release_group: N_l('Release Group Collections'),
+  series: N_lp('Series Collections', 'plural'),
+  work: N_l('Work Collections'),
+};
+
+const CollectionsTable = (({columns, data}) => {
```

Comment:
Just an observation: this component looks generic enough that it could be used for pretty much any table, not just collections. Maybe that's the (eventual) idea. :)

Reply:
Yes it is, see https://github.com/metabrainz/musicbrainz-server/pull/1246#issuecomment-545968391.